File: project-{topic}-enhanced/backend/src/entities/Project.ts
import { Entity, Column, ManyToOne, OneToMany } from 'typeorm';
import { BaseEntity } from './BaseEntity';
import { User } from './User';
import { Task } from './Task';

@Entity()
export class Project extends BaseEntity {
  @Column()
  name!: string;

  @Column({ type: 'text', nullable: true })
  description?: string;

  @ManyToOne(() => User, user => user.projects, { onDelete: 'CASCADE' })
  owner!: User;

  @Column()
  ownerId!: string; // Explicit column for ownerId

  @OneToMany(() => Task, task => task.project)
  tasks!: Task[];
}

File: project-{topic}-optimized/app/models/project.py
from sqlalchemy import Column, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base_class import Base
from sqlalchemy.dialects.postgresql import UUID

class Project(Base):
    name = Column(String, index=True, nullable=False)
    description = Column(Text, nullable=True)
    owner_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)

    owner = relationship("User", back_populates="projects")
    tasks = relationship("Task", back_populates="project", cascade="all, delete-orphan")

    def __repr__(self):
        return f"<Project(id={self.id}, name='{self.name}', owner_id='{self.owner_id}')>"
```